nix/process-compose/default.nix

```diff
@@ -45,17 +45,21 @@ in
           text = ''
             ${preHook}
 
-            run-process-compose () {
-              set -x; process-compose ${cliOutputs.global} --config ${configFile} "$@"; set +x
-            }
 
-            # Run `up` command, with arguments; unless the user wants to pass their own subcommand.
-            if [ "$#" -eq 0 ]; then
-              run-process-compose up ${cliOutputs.up}
-            else
-              run-process-compose "$@"
+            # If there are no arguments, it's the "up" command
+            # If the first argument is "up", it's also the "up" command
+            # If the first argument starts with a dash, we assume there isn't a subcommand, so it's also the "up" command
+            # Otherwise, we assume it's a subcommand other than "up"
+            params=(${cliOutputs.global})
+            set +u
+            if [ -z "$1" ] || [[ "$1" == "up" ]] || [[ "$1" == -* ]] ; then
```

Comment:
I don't see what purpose `[[ "$1" == -* ]]` serves here.

Reply:
I tried to explain it in the comments, but it looks like it failed 😅 

If the first argument starts with a dash, we know it's no subcommand. And if the first argument is no subcommand, we assume there is no subcommand argument at all. And if there is none, process-compose defaults to the up command. And therefore it's allowed to pass arguments dedicated to the up command.

Reply:
I understood that *what*, but I'm asking *why*.

> If the first argument starts with a dash, we assume there isn't a subcommand, so it's also the "up" command

This is of course wrong, since `--help` is not equivalent to `up`.

Reply:
Yes, that seems to be one of the edge cases I mentioned. Properly dealing with it could be tricky, though. Any ideas?

Reply:
Why not remove `[[ "$1" == -* ]]` entirely?